sale_order_partner_restrict/tests/test_sale_order_partner_restrict.py

```diff
@@ -0,0 +1,127 @@
+from odoo.tests import tagged
+from odoo.tests.common import SavepointCase
+
+
+@tagged("post_install", "-at_install")
+class TestSaleOrderPartnerRestrict(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.main_company = cls.env.ref("base.main_company")
+        cls.second_company = cls.env["res.company"].create({"name": "Second Company"})
+
+        cls.partner_model = cls.env["res.partner"]
+
+        cls.partner_parent = cls.partner_model.create(
+            {"name": "Partner Parent", "type": "contact"}
+        )
+        cls.partner_child = cls.partner_model.create(
+            {
+                "name": "Partner Child",
+                "type": "contact",
+                "parent_id": cls.partner_parent.id,
+            }
+        )
+        cls.partner_diff_company = cls.partner_model.create(
+            {
+                "name": "Partner Different Company",
+                "type": "contact",
+                "company_id": cls.second_company.id,
+            }
+        )
+        cls.partner_delivery = cls.partner_model.create(
+            {
+                "name": "Partner Delivery",
+                "type": "other",
+                "parent_id": cls.partner_parent.id,
+            }
+        )
+
+    def _create_sale_order(self):
+        so = self.env["sale.order"].create({"partner_id": self.partner_parent.id})
```

Comment:
I think here you can create the sales order like this

```suggestion
        so = self.env["sale.order"].create({"partner_id": self.partner_parent.id, "name": "/"})
```
The reason:

https://github.com/OCA/sale-workflow/blob/13.0/sale_order_revision/models/sale_order.py#L78